src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -57,17 +57,32 @@ after t (State value) = value (signal t) == newValue t
 
 type CircuitConcept a = Concept (State a) (Transition a) a
 
+flipSpec :: CircuitConcept a -> CircuitConcept a
+flipSpec c = mempty
+         {
+           initial = initial c,
+           arcs = flipCausalities,
+           interface = interface c,
+           invariant = invariant c
+         }
+  where
+    flipCausalities = map (\(f, t) -> createFlipCause f t) arcLists
+    createFlipCause f t = Causality (map toggle f) (toggle t)
+    arcLists = [ (f, t) | Causality f t <- (arcs c) ]
+
```

Comment:
I would do this a bit differently:
```haskell
-- This could be a useful primitive
dualCausality :: Causality a -> Causality a
dualCausality (Causality f t) = Causality (map toggle f) (toggle t)

-- then use it directly in dual
...
arcs = map dualCausality (arcs c)
...
```

Simpler and should work.